Order of name components no longer imposed

HelenaSabel opened the pull request:
This PR would close #325.

Deleting the templates for `forename` and `surname` (the solution presented in the Stylesheets meeting) was problematic on its own because it created a lot of whitespace problems in the outputs. The other changes in `common-core.xsl` propose a better handling of whitespace (and the addition of periods after the last author in bibliographic references).

I also made a couple of corrections in the test files because I consider that things like this (from `test.xml`) is wrong:
```
<name><forename>Charles</forename><surname>Dickens</surname></name>
```

I therefore added a space between the forename and the surname in this case, and also in `test27.xml` and in the bibliographic test of `Test2`. 

The test in `Test` concerning whitespace passed, but the one in `Test2` failed (see changes in https://github.com/TEIC/Stylesheets/commit/75fbb0824aacb6785edb047981c47e91e733e5d3).  I would like to ask the reviewers to pay special attention to this.

HelenaSabel committed:
no imposition of order in naming components and handling of whitespace

HelenaSabel committed:
no imposition of order in naming components and handling of whitespace

HelenaSabel committed:
whitespace correction in test files

HelenaSabel committed:
whitespace correction in test files

HelenaSabel committed:
order change in naming components

HelenaSabel committed:
order change in naming components

HelenaSabel committed:
whitespace changes: possible bug?

HelenaSabel committed:
whitespace changes: possible bug?

HelenaSabel committed:
changes in ids

HelenaSabel committed:
changes in ids

HelenaSabel committed:
whitespace corrected (due to change in test file)

HelenaSabel committed:
whitespace corrected (due to change in test file)

HelenaSabel committed:
irrelevant whitespace changes

HelenaSabel committed:
irrelevant whitespace changes

HelenaSabel committed:
changes in order of name components

HelenaSabel committed:
changes in order of name components

sydb commented:
> I also made a couple of corrections in the test files because I consider that things like this (from `test.xml`) is wrong:
> ```
> <name><forename>Charles</forename><surname>Dickens</surname></name>
> ```
That seems perfectly correct XML to me. (And I daresay, it is probably in test.xml like that just to make sure the Stylesheets do not screw up when a name is encoded like this.)

This encoding says “This is a name; it has two components, a forename and a surname; the forename is ‘Charles’; the surname is ‘Dickens’.”. It makes no assertion about what a processor should _do_ with that information. One quite reasonable approach is to output “Dickens, Charles”. Another is to output “Charles Dickens”. A third is to generate the URL “https://en.wikipedia.org/wiki/Charles_Dickens”. A fourth is to generate the personography key “./authors.xml#Dickens.Charles”. And of course, crunching it into “CharlesDickens” is a viable (if ugly) alternative, too.

HelenaSabel commented:
What concerns me about that is falling again in Western-centric practises, with the Stylesheets separating or reordering components following conventions that are not universal. That’s why I think a space in the source file is relevant because (in my opinion) it indicates that in the case of the name “Charles Dickens”, the name components are not agglutinated (which is not the case in other languages).

sydb commented:
_Really_ good point, but I do not think it invalidates the correctness of
~~~xml
<name><surname>石井</surname><forename>四郎</forename></name>
~~~
(Nor the correctness of
~~~xml
<name> <surname>Ishii</surname> <forename>Shirō</forename> </name>
~~~
nor of
~~~xml
<name>
  <surname>石井</surname>
  <pc force="strong">&#x20;</pc>
  <forename>四郎</forename>
</name>
~~~
.) I do not actually know a culture in which name components are agglutinated (Turkish? Finnish?), but certainly
~~~xml
<name>
  <surname>FAMILY</surname>
  <pc join="both"/>
  <forename>given</forename>
</name>
~~~
is worth considering.

Boils down to what used to be thought of as data-centric vs text-centered encoding, I guess.

All that said, just because `<name><forename>John</forename><surname>Lennon</surname></name>` is a perfectly reasonable encoding does not mean our Stylesheets have to process it.

HelenaSabel commented:
This PR goes back to draft, pending some improvements in the handling of bibliographic references

ebeshero commented:
Council at VF2F 16 March 2024: We recognize that the changes here are definitely an improvement for a more culturally sensitive processing of parts of a name based on the source encoding. @joeytakeda suggests we should keep the option to access the original template. For users who work with teiGarage, we should be sure this kind of processing is available to them at command line.

@HelenaSabel should update the branch and then request reviewers look at this again to be sure it's okay to merge, and to decide whether to allow the option to access the original template.

rettinghaus committed:
fix: update link in README.md

rettinghaus committed:
Rephrase

rettinghaus committed:
update list types to use rend

rettinghaus committed:
update test2 files

rettinghaus committed:
updated test files

rettinghaus committed:
update docs

rettinghaus committed:
fix tests

trishaoconnor committed:
Update link to TCW 27

trishaoconnor committed:
Update link to TCW 27

martindholmes commented:
I have worked on a project which mixed Icelandic names with other primarily European and North American names in reference lists, and required that they be processed/output differently; eventually, we defaulted to encoding the names in the correct form/order and with any punctuation required, and outputting them as-is. To retain the original behaviour, as @joeytakeda suggests, without causing undue offence, it perhaps the `<name>` / `<persName>` elements here should have `@xml:lang` attributes, or look upwards for the nearest `@xml:lang`, and be processed accordingly. So a name with "en" might be processed into "surname, forename" while a name with "jp" would go to "forenamesurname", and a name with "is" would go to "forename surname".

jbampton committed:
misc: fix spelling

rvdb committed:
- switched output to OpenEdition-1.6.4
- added support for //titleStmt/(editor|author)/idno

rvdb committed:
copy *:xsl children of //tei:constraintDecl[@scheme eq 'schematron'] to RNG

trishaoconnor committed:
Merge pull request #762 from rvdb/rvdb_copy_xsl_constraintDecl_children_to_rng

copy *:xsl children of //tei:constraintDecl[@scheme eq 'schematron'] …

joeytakeda committed:
Merge pull request #761 from rvdb/jTEI_OpenEdition-1.6.4

- switched output to OpenEdition-1.6.4

joeytakeda committed:
Merge pull request #747 from TEIC/trishaoconnor_694

Update target link to TCW 27 for deprecation warnings

ebeshero committed:
Merge pull request #710 from rettinghaus/patch-1

fix #703: update link in README.md

martindholmes reviewed (approved):
The result of the changes appears to be to output:

surname forename (i.e. Holmes Martin)

without a comma, in e.g. test27, when the original encoding has surname forename with no punctuation, rather than reversing the order as was previously done. I believe this is the correct thing to do -- in other words, it's right to output names in the order encoded and with any punctuation that is in the encoding, not supplying any -- but I expect this might take one or two people by surprise.

HelenaSabel committed:
adjust version number in preparation for release

bleekere committed:
update p5 subset with HBS

sydb committed:
Remove extraneous file noticed
by @peterstadler

bleekere committed:
Update changelog

bleekere committed:
increment version number

sydb committed:
Grab latest from dev

sydb committed:
A proposed fix for
#773. I am quite confident it fixes the problem identified
in that issue, but does it break anything else?

martindholmes committed:
Deletion of unused file per issue #187.

trishaoconnor committed:
Merge pull request #775 from TEIC/issue-187-unused-file

Deletion of unused file per issue #187.

sydb committed:
Revert spelling corrrections that
are in other people’s files.

trishaoconnor committed:
Merge pull request #768 from TEIC/sydb_438_rm_superflous_file

Remove extraneous file

trishaoconnor committed:
Merge pull request #760 from jbampton/fix-typos

misc: fix spelling

trishaoconnor committed:
Merge pull request #774 from TEIC/sydb_773_min_0_max_2_or_more

fix for `@minOccurs` of 0 and `@maxOccurs` of 2 or more (but not "unbounded")

rvdb committed:
improved presentation of multilingual keyword lists in different languages

rvdb committed:
Revert "improved presentation of multilingual keyword lists in different languages"

This reverts commit eeaf51eb8564bb937c29b3ab9b932d2c39765f5c.

sydb committed:
update P5 subset

sydb committed:
Missed one

trishaoconnor committed:
Merge pull request #745 from rettinghaus/update_lists

fix #734: Update lists

rvdb committed:
Merge remote-tracking branch 'upstream/dev' into dev

rvdb committed:
added zero-width space before &amp; in URLs that are displayed in the running text, in order to create break-points for the XSL-FO formatter

martindholmes committed:
Merge pull request #785 from rvdb/jTEI-splitLongURLSinPDF

jTEI: split long urls in pdf

HelenaSabel committed:
merge
